### PR TITLE
Raise for unsupported engine and pop from kwargs

### PR DESCRIPTION
**Pull request by phofl**
keeping the engine in the kwargs passes it along to arrow which raises

### CODE REVIEW

File: dask_expr/io/tests/test_io.py [hunk 1]
```diff
@@ -328,6 +328,13 @@ def test_to_parquet(tmpdir, write_metadata_file):
         df2.to_parquet(tmpdir, overwrite=True)
 
 
+def test_to_parquet_engne(tmpdir):
```
Comment (rjzamora) [inline dask_expr/io/tests/test_io.py:331]:
```suggestion
def test_to_parquet_engine(tmpdir):
```

File: dask_expr/io/parquet.py [hunk 2]
```diff
@@ -192,7 +192,11 @@ def to_parquet(
     from dask_expr._collection import new_collection
     from dask_expr.io.parquet import NONE_LABEL, ToParquet
 
-    engine = _set_parquet_engine(meta=df._meta)
+    engine = kwargs.pop("engine")
+    if engine == "fastparquet":
+        raise NotImplementedError("Fastparquet engine is not supported")
+
+    engine = _set_parquet_engine(engine=engine, meta=df._meta)
```
Comment (rjzamora) [inline dask_expr/io/parquet.py:199]: This seems fine. We could also include `engine=None` in the explicit argument list above and/or to move the "fastparquet" check into `_set_parquet_engine`.

**Review by rjzamora — APPROVED** [review]

File: dask_expr/io/parquet.py [hunk 3]
```diff
@@ -192,7 +192,11 @@ def to_parquet(
     from dask_expr._collection import new_collection
     from dask_expr.io.parquet import NONE_LABEL, ToParquet
 
-    engine = _set_parquet_engine(meta=df._meta)
+    engine = kwargs.pop("engine")
```
Comment (rjzamora) [inline dask_expr/io/parquet.py:195]:
```suggestion
    engine = kwargs.pop("engine", None)
```

File: dask_expr/io/parquet.py (hunk 2 above)
Comment (phofl) [reply dask_expr/io/parquet.py]: I like this

**Pull request merged**